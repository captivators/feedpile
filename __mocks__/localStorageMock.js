const localStorageMock = (() => {
  let store = {
    "profile": '{"name":"faiz"}',
    id_token: ""
  };
  return {
    getItem(key) {
      return store[key]
    },
    setItem(key, value) {
      store[key] = value.toString()
    },
    clear() {
      store = {}
    }
  };
})();

global.localStorage = localStorageMock;
