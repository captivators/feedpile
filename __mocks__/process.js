const process = {
  "env" : {
    "AUTH_ID":"test",
    "AUTH_DOMAIN":"user_id"
  }
}
global.process = process;
