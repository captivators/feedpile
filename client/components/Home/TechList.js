import React from 'react';
import TechItem from './TechItem';


const style = {
  logoList: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    flexWrap: 'wrap',
  },
  first: {
    paddingTop: '45px',
    paddingBottom: '45px',
    width: '100%',
    textAlign: 'center',
    h1: {
      color: '#fff',
      marginTop: 0,
      fontSize: '48px',
    }
  }
}


const techList = [
  {
    link: 'http://redux.js.org/',
    image: './images/redux.png'
  },
  {
    link: "http://redux.js.org/",
    image: "./images/webpack.png"
  },
  {
    link: "http://nodejs.org/eng",
    image: ""
  }
]

const TechList = (props) => {
  return (
    <div>
      <div style={style.first}>
        <h2>The Tech </h2>
      </div>
      <div className="stack" style={style.logoList}>
        {techList.map((tech, index) => (
          <TechItem tech= { tech } key={ index } />
        ))}
      </div>
    </div>
  )
}

export default TechList;