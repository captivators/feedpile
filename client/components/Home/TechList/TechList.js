import React from 'react';
import TechItem from '../TechItem/TechItem';

const style = {
  logoList: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    boxSizing: 'border-box',
    flexWrap: 'wrap',
    backgroundColor: '#78909C',
    padding: '40px',
  }
}


const techList = [
  {
    name: "React",
    link: "https://facebook.github.io/react/",
    image: "https://s27.postimg.org/ud14uv8v7/logo-578x270.png"
  },
  {
    name: "React Router",
    link: "https://react-router.js.org",
    image: "https://s18.postimg.org/7c5q3p115/react-router-logo-_AB5_BFB638_F-seeklogo.com.png"
  },
  {
    name: "Redux",
    link: 'http://redux.js.org/',
    image: "https://s30.postimg.org/50zpvkri9/logo-redux.png"
  },
  {
    name: "Webpack",
    link: "https://www.webpack.js.org/",
    image: "https://s15.postimg.org/p04co7pej/webpack.png",
  },
  {
    name: "Babel",
    link: "http://www.babeljs.io/",
    image: "https://s28.postimg.org/59tbpai4t/babel_1.png"
  },
  {
    name: "Node",
    link: "http://nodejs.org/eng",
    image: "https://s1.postimg.org/400bfw3vj/nodejs.png"
  },
  {
    name: "Express",
    image: "https://s7.postimg.org/c5kgs8wh7/express.png",
    link: "https://expressjs.com"
  },
  {
    name: "MongoDB",
    link: "https://www.mongodb.com",
    image: "https://s24.postimg.org/hx7jao6qd/mongodb-logo-1.png",
  },
  {
    name: "Travis CI",
    link: "https://travis-ci.org",
    image: "https://s23.postimg.org/n1i0g8re3/Travis_CI-_Mascot-1-20feeadb48fc2492ba741d89cb5a5c.png"
  },
  {
    name: "Docker",
    link: "https://www.docker.com/what-docker",
    image: "https://s23.postimg.org/iqwvdr6mz/docker.png",
  },
  {
    name: "AWS",
    link: "https://aws.amazon.com/codedeploy/",
    image: "https://s15.postimg.org/c0rtqswi3/Amazon_Webservices_Logo.svg.png",
  },
  {
    name: "Jest",
    link: '',
    image: "https://s24.postimg.org/rgds3co7p/jest.9ec4c2ad.png"
  },
  {
    name: "Enzyme",
    link: "http://airbnb.io/enzyme/docs/api/",
    image: "https://s7.postimg.org/p125vwx4r/1-pu9_U8_EYL3_KGrgvapyp1p_Sg.png"
  },
  {
    name: "Mocha Chai",
    link: "http://chaijs.com/",
    image: "https://s9.postimg.org/q5c35f0zj/chaijs-mocha.png"
  },
  {
    name: "Material UI",
    link: "http://www.material-ui.com/#/customization/themes",
    image: "https://s11.postimg.org/975jiaz8j/material-ui_1.png"
  },
  {
    name: "NPM",
    link: "https://www.npmjs.com/",
    image: "https://s14.postimg.org/nhna7dump/npm.png"
  }
]

const TechList = (props) => {
  return (
    <div className="stack" style={style.logoList}>
      {techList.map((tech, index) => (
        <TechItem tech= { tech } key={ index } />
      ))}
    </div>
  )
}

export default TechList;
