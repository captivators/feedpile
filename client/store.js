import {createStore, compose, applyMiddleware} from 'redux';
import rootReducer from './root-reducer';
import logger from 'redux-logger';
import createSagaMiddleware from 'redux-saga';
import rootSaga from './root-saga';

const sagaMiddleware = createSagaMiddleware();

const store = createStore(rootReducer, compose (
  applyMiddleware(logger, sagaMiddleware)
));

sagaMiddleware.run(rootSaga);
export default store;
