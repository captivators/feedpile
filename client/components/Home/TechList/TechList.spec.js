import React from 'react';
import { shallow, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import '../../../../__mocks__/localStorageMock';
import {Provider} from 'react-redux';
import store from '../../../store';
import TechList from './TechList'

test('Landingpage: Team section snapshot test', () => {

  const component = shallow(
      <TechList />
  );
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

test('Landingpage: TechList section should render 15 tech images', () => {
  const component = render(<Provider store={store}><TechList /></Provider>);
  expect(component.find('.stack').length).toEqual(1);
  expect(component.find('.tech-logo').length).toEqual(16);
})