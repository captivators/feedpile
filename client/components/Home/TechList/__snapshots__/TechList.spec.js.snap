// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landingpage: Team section snapshot test 1`] = `
<div
  className="stack"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#78909C",
      "boxSizing": "border-box",
      "display": "flex",
      "flexWrap": "wrap",
      "justifyContent": "center",
      "padding": "40px",
    }
  }
>
  <TechItem
    tech={
      Object {
        "image": "https://s27.postimg.org/ud14uv8v7/logo-578x270.png",
        "link": "https://facebook.github.io/react/",
        "name": "React",
      }
    }
  />
  <TechItem
    tech={
      Object {
        "image": "https://s18.postimg.org/7c5q3p115/react-router-logo-_AB5_BFB638_F-seeklogo.com.png",
        "link": "https://react-router.js.org",
        "name": "React Router",
      }
    }
  />
  <TechItem
    tech={
      Object {
        "image": "https://s30.postimg.org/50zpvkri9/logo-redux.png",
        "link": "http://redux.js.org/",
        "name": "Redux",
      }
    }
  />
  <TechItem
    tech={
      Object {
        "image": "https://s15.postimg.org/p04co7pej/webpack.png",
        "link": "https://www.webpack.js.org/",
        "name": "Webpack",
      }
    }
  />
  <TechItem
    tech={
      Object {
        "image": "https://s28.postimg.org/59tbpai4t/babel_1.png",
        "link": "http://www.babeljs.io/",
        "name": "Babel",
      }
    }
  />
  <TechItem
    tech={
      Object {
        "image": "https://s1.postimg.org/400bfw3vj/nodejs.png",
        "link": "http://nodejs.org/eng",
        "name": "Node",
      }
    }
  />
  <TechItem
    tech={
      Object {
        "image": "https://s7.postimg.org/c5kgs8wh7/express.png",
        "link": "https://expressjs.com",
        "name": "Express",
      }
    }
  />
  <TechItem
    tech={
      Object {
        "image": "https://s24.postimg.org/hx7jao6qd/mongodb-logo-1.png",
        "link": "https://www.mongodb.com",
        "name": "MongoDB",
      }
    }
  />
  <TechItem
    tech={
      Object {
        "image": "https://s23.postimg.org/n1i0g8re3/Travis_CI-_Mascot-1-20feeadb48fc2492ba741d89cb5a5c.png",
        "link": "https://travis-ci.org",
        "name": "Travis CI",
      }
    }
  />
  <TechItem
    tech={
      Object {
        "image": "https://s15.postimg.org/c0rtqswi3/Amazon_Webservices_Logo.svg.png",
        "link": "https://aws.amazon.com/codedeploy/",
        "name": "AWS",
      }
    }
  />
  <TechItem
    tech={
      Object {
        "image": "https://s23.postimg.org/iqwvdr6mz/docker.png",
        "link": "https://www.docker.com/what-docker",
        "name": "Docker",
      }
    }
  />
  <TechItem
    tech={
      Object {
        "image": "https://s24.postimg.org/rgds3co7p/jest.9ec4c2ad.png",
        "link": "",
        "name": "Jest",
      }
    }
  />
  <TechItem
    tech={
      Object {
        "image": "https://s7.postimg.org/p125vwx4r/1-pu9_U8_EYL3_KGrgvapyp1p_Sg.png",
        "link": "http://airbnb.io/enzyme/docs/api/",
        "name": "Enzyme",
      }
    }
  />
  <TechItem
    tech={
      Object {
        "image": "https://s9.postimg.org/q5c35f0zj/chaijs-mocha.png",
        "link": "http://chaijs.com/",
        "name": "Mocha Chai",
      }
    }
  />
  <TechItem
    tech={
      Object {
        "image": "https://s11.postimg.org/975jiaz8j/material-ui_1.png",
        "link": "http://www.material-ui.com/#/customization/themes",
        "name": "Material UI",
      }
    }
  />
  <TechItem
    tech={
      Object {
        "image": "https://s14.postimg.org/nhna7dump/npm.png",
        "link": "https://www.npmjs.com/",
        "name": "NPM",
      }
    }
  />
</div>
`;
